### ISSUE TITLE
[Feature Request]: RabbitMQ IO connector for python?

### ISSUE THREAD

**Issue by nitishxp** [apache/beam#26393]
### What would you like to happen?

 I am finding that we don't have rabbitmq io connector for python is there any plan for to be available in python?

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by tvalentyn** [comment]
@johnjcasey @chamikaramj 
Is it possible to use Java's RabbitMQ IO via XLang ?

**Comment by nitishxp** [comment]
can we use it using xlang? @johnjcasey  @chamikaramj

**Comment by chamikaramj** [comment]
It's definitely possible to add support for it via x-lang but we don't have a Python x-lang wrapper for this yet.

**Comment by nitishxp** [comment]
any timeline when that will be added :)

**Comment by chamikaramj** [comment]
I don't think we have an ETA for this currently but contributions are welcome.

**Comment by nitishxp** [comment]
@chamikaramj  I haven't worked on java before nor have any knowledge of xlang could you point me a sample in beam repo where x-lang has been used?

**Comment by hamza-digitl** [comment]
@chamikaramj, I also need this IO connector, do you have any news about its integration?
@nitishxp did you find any workarounds?

**Comment by chamikaramj** [comment]
We recently started auto-generating Python wrappers for Java schema-transforms. So the work needed here should boil down to developing a schema-aware transforms for Java Rabbit MQ read/write transforms. Basically we have to add implementations of the two abstract classes given below. @ahmedabu98 do we have better documentation on implementing new schema-aware transforms ?

[1] https://beam.apache.org/releases/javadoc/current/org/apache/beam/sdk/schemas/transforms/SchemaTransformProvider.html
[2] https://beam.apache.org/releases/javadoc/current/org/apache/beam/sdk/schemas/transforms/SchemaTransform.html

**Comment by ahmedabu98** [comment]
Hey, I was working on releasing a couple of blogs a few months ago but the work got halted cuz of other priorities. The drafts in their current state are still comprehensive though.

Please take a look here for how to develop a SchemaTransform ([WIP blog](https://docs.google.com/document/d/1h7q52Iq01U1uVdVFFSsZVmR_SMnJloQL-HXMQjqqPXU/edit?resourcekey=0-BvaeEeUT5uCvzpUGOe29wg#heading=h.17ub1jvvdl8l))
And take a look here for how to use existing Python utilities to generate a wrapper for this transform: ([WIP blog](https://docs.google.com/document/d/1kaIdFXv0lf7VKd5epUdyYAA98gEXrJTaFyvo5zY6OT8/edit#heading=h.l4mqhwd7gvog))
